### PR TITLE
[Darkside] Theme component as feature-flagging

### PR DESCRIPTION
**Pull request by KenAJoh**
### Description

Allows the use of useAkselTheme to check if the new theme is in use and thus do some markup-changes in components

### Component Checklist 📝

- [ ] JSDoc
- [ ] Examples
- [ ] Documentation / Decision Records
- [ ] Storybook
- [ ] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [ ] Component tokens (`@navikt/core/css/tokens.json`)
- [ ] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [ ] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [ ] New component? CSS import (`@navikt/core/css/index.css`)
- [ ] Breaking change? Update migration guide. Consider codemod.
- [ ] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

File: @navikt/core/react/src/provider/theme/AkselTheme.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import cl from "clsx";
+import React, { forwardRef } from "react";
+import { createContext } from "../../util/create-context";
+
+type AkselThemeContext = {
+  appearance: "light" | "dark";
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/provider/theme/AkselTheme.tsx]: Why not call it theme?
```suggestion
  theme: "light" | "dark";
```

File: @navikt/core/react/src/provider/theme/AkselTheme.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+const [ThemeProvider, useAkselTheme] = createContext<AkselThemeContext>({
+  errorMessage: "useAkselThemeProvider must be within AkselThemeProvider",
+  hookName: "useAkselThemeProvider",
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/provider/theme/AkselTheme.tsx]: We usually have "Context" as suffix in the hook name...
```suggestion
  errorMessage: "useAkselThemeContext must be used within AkselThemeProvider",
  hookName: "useAkselThemeContext",
```
...or no suffix:
```suggestion
  errorMessage: "useAkselTheme must be used within AkselThemeProvider",
  hookName: "useAkselTheme",
```

File: @navikt/core/react/src/provider/theme/AkselTheme.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      hasBackgroundProp ?? (isRoot || props.appearance !== undefined);
+
+    return (
+      <ThemeProvider appearance={appearance}>
+        <div
+          ref={ref}
+          className={cl("navds-theme", className, {
+            light: appearance === "light",
+            dark: appearance === "dark",
+          })}
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/provider/theme/AkselTheme.tsx]: Could we just do this, or could that be a problem? 🤔
```suggestion
          className={cl("navds-theme", className, appearance)}
```

File: @navikt/core/react/src/provider/theme/AkselTheme.tsx (hunk 1 above)
Comment (KenAJoh) [reply @navikt/core/react/src/provider/theme/AkselTheme.tsx]: Im a little conflicted by what's the "most" correct naming for these cases:
- We have light and dark-mode
- We add support for High contrast (or some other)

What is the "theme" now? I could argue that we are using a "high-contrast"-theme with a dark "appearance". But might just be overcomplicating it. For now i changed it to "theme".

File: @navikt/core/react/src/provider/theme/AkselTheme.tsx (hunk 3 above)
Comment (KenAJoh) [reply @navikt/core/react/src/provider/theme/AkselTheme.tsx]: Might make it a little obscure for what's happening, but did it for now 👍

File: @navikt/core/react/src/provider/theme/AkselTheme.tsx (hunk 1 above)
Comment (HalvorHaugan) [reply @navikt/core/react/src/provider/theme/AkselTheme.tsx]: Let's cross that bridge when we get to it 😄

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**